spec/src/main/asciidoc/architecture.asciidoc

```diff
@@ -1199,3 +1199,108 @@ The following metrics are produced for each channel declared by the application
 // ==== Custom message types
 
 // Implementations may provide custom subtypes of `Message` as an end user API, for the purpose of providing additional metadata associated with the messaging provider. If an application declares that it consumes or produces a subtype of the `Message` wrapper, the implementation should validate that the messaging provider supports that message wrapper type, and raise a deployment exception if it doesn't.
+
+== Use Reactive Messaging in the imperative world
+Traditionally, the reactive world and imperative world are separated and operate in parallel. Reactive Messaging deals with streams of data in the reactive world, while the imperative world is pretty much point to point communication. However, imperative programme sometimes needs to connect to reactive streams so that responses can be emitted to a destination service. 
+It would be nice if Reactive Messaging is available to the imperative world. This section is about making Reactive Messaging available to the imperative world.
```

Comment:
Not sure about "It would be nice if Reactive Messaging is available to the imperative world."

Reply:
it's so nice that we did it :-D

Reply:
reworded